From 0d0f21abe4de201b2d0baa05b487c1a335bc4320 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 13 Feb 2025 10:51:23 +0800
Subject: [PATCH] GENTOO: sip: add a missing include to QVariant

This was removed between 6.8.0.dev2411020930 and 6.8.0 final, but
results in a build failure if missing (or at least with Qt 6.8.1).

qwebengineframe.sip:52:64: error: invalid use of incomplete type 'class QVariant'

Signed-off-by: Ionen Wolkens <ionen@gentoo.org>

Link: https://gitweb.gentoo.org/repo/gentoo.git/commit/dev-python/pyqt6-webengine/files/pyqt6-webengine-6.8.0-qvariant.patch?id=4fea1dcf05c5db029b96bb6a99379f42e7c18d09
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 sip/QtWebEngineCore/qwebengineframe.sip | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sip/QtWebEngineCore/qwebengineframe.sip b/sip/QtWebEngineCore/qwebengineframe.sip
index d5420e6..9bdf528 100644
--- a/sip/QtWebEngineCore/qwebengineframe.sip
+++ b/sip/QtWebEngineCore/qwebengineframe.sip
@@ -28,6 +28,10 @@ class QWebEngineFrame /NoDefaultCtors/
 #include <qwebengineframe.h>
 %End
 
+%TypeCode
+#include <QVariant>
+%End
+
 public:
     bool isValid() const;
     QString name() const;
-- 
2.48.1

